### PR TITLE
automatically save adjustments to offsets of equipped items

### PR DESCRIPTION
**Pull request by sethalves**
- automatically save adjustments to offsets of equipped items

### CODE REVIEW

**Comment by sethalves** [comment]
to test: run the handControllerGrab.js from this PR as well as attachmentManager.js.  equip something in one hand and adjust its position with the other.  drop the entity and then re-equip it and see that the position snaps back to where it was before it was dropped.

**Comment by hifi-gustavo** [comment]
Build console output for commit: e4ef3ae8697f8431c8f4191a43eda3b3b2f0d289 available at: https://gist.github.com/e78491b53059f8be0143<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.dmg

**Comment by cozza13** [comment]
-default spot for the gun, block etc is off the back end of the wrist.
-can not do the adjustments with the bow and arrow.
- object does not collide with an object when it is picked up. e.g. hold a block and try and put another one on top
- I have had situations where the gun is not going back to the same spot.

It is a nice feeling to adjust

**Comment by sethalves** [comment]
The updated entities in demo should have fixed offsets.  We wont be able to adjust the bow, because it disables the other hand when equipped.

To test wearing things, put something like:

{"grabbableKey":{"grabbable":true},"wearable":{"joints":{"Head":[]}}}

in the userData of an entity.  Pick it up and drop it near your head.

**Comment by nbq** [comment]
When I pick something up with the Hydras, I never get it to lock to a position, even on demo. I have to use the other hand to sort of move it into place a bit then it locks into the spot. That is one of the problems I have with grabbing and the Hydras is that nothing I pick up ever locks into place even if I pick it up with the hand its supposed to lock to

**Comment by hifi-gustavo** [comment]
Build console output for commit: 30e97c2a76bcdafa30d2f2cef9866917de680fc6 available at: https://gist.github.com/e78491b53059f8be0143<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.dmg

**Comment by cozza13** [comment]
I have tested this and the feature is working. You do need to be running both scripts.

**Comment by sethalves** [comment]
I added attachedEntitiesManager.js to default scripts.

File: examples/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -1217,10 +1217,12 @@ function MyController(hand) {
...
-        if (handJointName in allowedJoints) {
-            return true;
+        if ("joints" in wearableData) {
+            var allowedJoints = wearableData.joints;
+            var handJointName = this.hand === RIGHT_HAND ? "RightHand" : "LeftHand";
+            if (handJointName in allowedJoints) {
+                return true;
+            }
         }
     }
```
Comment (AndrewMeadows) [inline examples/controllers/handControllerGrab.js]: This function has no **return** statement outside the new **if** statements.  Maybe it should **return false** at the end?

File: examples/defaultScripts.js [hunk 2]
```diff
@@ -21,3 +21,4 @@ Script.load("controllers/squeezeHands.js");
 Script.load("grab.js");
 Script.load("directory.js");
 Script.load("dialTone.js");
+Script.load("attachedEntitiesManager.js");
```
Comment (AndrewMeadows) [inline examples/defaultScripts.js]: If we keep  _attachedEntitiesManager.js_ on the list of default scripts then we will have to eventually cleanup the assets to NOT use links from @sethalves personal assets (e.g. http://headache.hungry.com/~seth/hifi/save.png).  So the question is:  Do we want to clean these up now or later?

File: examples/attachedEntitiesManager.js [hunk 3]
```diff
@@ -22,41 +22,44 @@ var MINIMUM_DROP_DISTANCE_FROM_JOINT = 0.4;
 var ATTACHED_ENTITY_SEARCH_DISTANCE = 10.0;
 var ATTACHED_ENTITIES_SETTINGS_KEY = "ATTACHED_ENTITIES";
 var DRESSING_ROOM_DISTANCE = 2.0;
+var SHOW_TOOL_BAR = false;
```
Comment (AndrewMeadows) [inline examples/attachedEntitiesManager.js]: **SHOW_TOOL_BAR** is hardcoded as **false**.  Does this mean the toolbar is a 100% debug thing?

Comment (sethalves) [inline examples/attachedEntitiesManager.js]: Yes, for now.  The script wont have a UI once it's finished.

File: examples/defaultScripts.js (hunk 2 above)
Comment (sethalves) [inline examples/defaultScripts.js]: I've removed the hungry.com bit.

File: examples/controllers/handControllerGrab.js (hunk 1 above)
Comment (sethalves) [inline examples/controllers/handControllerGrab.js]: hm... I suppose javascript was accidently doing the right thing, here.  fixed

**Comment by hifi-gustavo** [comment]
Build console output for commit: ad2a7bfc3c4048bae461a0de08244ab0108c92fc available at: https://gist.github.com/e78491b53059f8be0143<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7003.dmg

**Pull request merged**